staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -816,6 +816,38 @@ func (c *Cacher) dispatchEvents() {
 	}
 }
 
+func setCachingObjects(event *watchCacheEvent, versioner storage.Versioner) {
+	if object, err := newCachingObject(event.Object); err == nil {
```

Comment:
event.Object can be nil for delete, so only wrap if non-nil (or handle nil in `newCachingObject`)

Reply:
Actually - it is never nil. I added a comment.